chia/wallet/wallet_state_manager.py

```diff
@@ -1130,6 +1130,8 @@ async def handle_clawback(
         # Record metadata
         assert coin_state.created_height is not None
         is_recipient: Optional[bool] = None
+        # For resync correctly, we need to fetch the coin state again
+        latest_coin_state = (await self.wallet_node.get_coin_state([coin_state.coin.name()], peer=peer))[0]
```

Comment:
Re-fetching the coin we are currently processing doesn't look right. We would we need to do that?

Reply:
This is for fixing the resync issue of clawback. After the resync, the wallet may not update the clawback coin to spent, which causes the related clawback tx also be stuck in pending status. The possible root cause is the concurrent syncing. We may process or fetch the higher blocks before the block which confirms the clawback coin. To solve this, I added an additional check to fetch the latest coin state. The passed-in coin state is for a certain height.
This fix is verified on the testnet.

Reply:
But why do you need to fetch the coins state again here? Wouldn't it be enough to just do what you do below with `coin_state` instead of `latest_coin_state`? If not, i think it would be good to investigate what is really happening here in the failure case and maybe try to reproduce this issue in a test?

Reply:
Since the coin_state is always unspent, the coin may already have been spent long time ago. This is a general issue for syncing and caused many weird issues, not just for Clawback. We should think about if it is feasible to resync synchronously. I mean only process the higher blocks before all previous blocks are processed.

Reply:
I just added a log. Seems the coin_state has a chance to be set as spent. So the question is why sometimes the coin_State is spent but sometimes it doesn't.

Reply:
This is removed